Assets/Scripts/Entity/Enemy/EnemyState.cs

```diff
@@ -41,7 +41,9 @@ private void Update()
 
         public override void Die()
         {
-            base.Die();
+            if (IsDead)
```

Comment:
I'd prefer it if you'd move this to the base method and keep `base.Die()` at the top here, since the check should be executed for all `Damageables` :)

Reply:
Can do, but then I have to find a nice way to tell the extended classes to skip as well

Reply:
Can do, but then I have to tell the extended classes to skip as well and I don't know how to do that nicely

Reply:
Okay, nvm, didn't think that one through 😅